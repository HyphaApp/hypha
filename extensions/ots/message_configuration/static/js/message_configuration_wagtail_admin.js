"use strict";

(function ($) {
    function add_update_events() {
        $(".message_type").each(function () {
            var message_type = $(this);
            var textarea = message_type
                .siblings(".email_message")
                .find("textarea");
            var select = message_type.find("select");
            if (!select.attr("evented")) {
                select.attr("evented", true);
                select.change(function () {
                    var message_id = $(this).val();
                    var message = $(
                        ".messaging_help_panel [attr-email-id='" +
                            message_id +
                            "']"
                    )
                        .text()
                        .trim();
                    textarea.val(message);
                });
            }
        });
    }

    $(function () {
        $("#id_messaging_settings-ADD").click(function () {
            add_update_events();
        });
        add_update_events();
    });
})(jQuery);
