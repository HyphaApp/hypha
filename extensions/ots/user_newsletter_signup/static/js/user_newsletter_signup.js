"use strict";

(function ($) {
    $(function () {
        $(".profile_newsletter_signup").click(function () {
            window.alert("You clicked me!");
        });
    });
})(jQuery);
