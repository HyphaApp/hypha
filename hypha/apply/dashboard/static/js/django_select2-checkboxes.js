(function ($) {
    $.fn.select2.amd.require(
        [
            "select2/multi-checkboxes/selection",
            "select2/multi-checkboxes/results",
        ],
        function (SelectionAdapter, ResultsAdapter) {
            $(function () {
                $(".django-select2-checkboxes").each(function (i, element) {
                    var $element = $(element);
                    $element.select2({
                        placeholder: $element.data("placeholder"),
                        closeOnSelect: false,
                        templateSelection: function (data) {
                            let filterType = $element.data("placeholder");

                            if (!data.selected.length) {
                                return filterType;
                            } else if (
                                data.selected.length == data.all.length
                            ) {
                                return "All " + filterType + " selected";
                            }
                            return (
                                data.selected.length +
                                " of " +
                                data.all.length +
                                " " +
                                filterType
                            );
                        },
                        selectionAdapter: SelectionAdapter,
                        returnesultsAdapter: ResultsAdapter,
                    });
                });
            });
        }
    );
})(this.jQuery);
