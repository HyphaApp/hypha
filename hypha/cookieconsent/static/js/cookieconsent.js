(function () {
    "use strict";

    if (typeof Cookies !== "undefined") {
        const cookieconsent = document.querySelector(".cookieconsent");

        if (
            typeof Cookies.get("cookieconsent") === "undefined" &&
            cookieconsent
        ) {
            cookieconsent.classList.add("js-cookieconsent-open");
        }

        const cookie_buttons = Array.prototype.slice.call(
            document.querySelectorAll("button[data-consent]")
        );
        const sitedomain = window.location.hostname.split(".").slice(-2);
        const cookiedomain = sitedomain.join(".");
        let cookie_options = [];
        cookie_options["domain"] = cookiedomain;
        cookie_options["sameSite"] = "strict";
        cookie_options["expires"] = 365;
        if (window.location.protocol === "https:") {
            cookie_options["secure"] = true;
        }

        cookie_buttons.forEach(function (button) {
            button.addEventListener("click", function () {
                if (button.getAttribute("data-consent") == "true") {
                    Cookies.set("cookieconsent", "accept", cookie_options);
                } else {
                    Cookies.set("cookieconsent", "decline", cookie_options);
                }
                cookieconsent.classList.remove("js-cookieconsent-open");
            });
        });
    }
})();
