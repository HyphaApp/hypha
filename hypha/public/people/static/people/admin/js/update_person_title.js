$(document).ready(function () {
    var $lastNameInput = $("#id_last_name");
    var $firstNameInput = $("#id_first_name");
    var $titleInput = $("#id_title");
    var $slugInput = $("#id_slug");

    $firstNameInput.on("input", function () {
        joinFirstNameLastName();
    });

    $lastNameInput.on("input", function () {
        joinFirstNameLastName();
    });

    function joinFirstNameLastName() {
        var firstName = $firstNameInput.val();
        var lastName = $lastNameInput.val();
        var title = firstName + " " + lastName;

        $slugInput.data("previous-val", $slugInput.val());
        $titleInput.data("previous-val", $titleInput.val());
        $titleInput.val(title);
        $titleInput.blur(); // Trigger slug update
    }
});
