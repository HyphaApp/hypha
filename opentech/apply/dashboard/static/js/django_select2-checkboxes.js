(function ($) {
    var init = function ($element) {
        options = {
            placeholder: $element.data('placeholder'),
            templateSelection: function(selected, total) {
                let filterType = this.placeholder;
                if ( !selected.length ) {
                    return filterType;
                } else if ( selected.length===total ) {
                    return 'All ' + filterType + ' selected';
                }
                return selected.length + ' of ' + total + ' ' + filterType + ' selected';
            }
        };
        $element.select2MultiCheckboxes(options);
    };
    $(function () {
        $('.django-select2-checkboxes').each(function (i, element) {
            var $element = $(element);
            init($element);
        });
    });
}(this.jQuery));
