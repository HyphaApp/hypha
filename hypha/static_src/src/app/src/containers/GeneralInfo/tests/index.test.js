import React from "react";
import {mount} from "enzyme";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import initialState from '../models'
import  {GeneralInfoContainer}  from '../index.js';

enzyme.configure({adapter: new Adapter()});

const mockStore = configureMockStore();
describe("Test General info Container", () => {
    let store;
    const locale = "en-US";
    it("Should render General info without issues", () => {
        store = mockStore({
            "Settings": initialState
        });
        const initializeAction = jest.fn()
        const wrapper = mount(
            <Provider
                store={store}
            >
                <GeneralInfoContainer initializeAction={initializeAction}/>
            </Provider>
        );
        expect(wrapper.find('.container').length).toEqual(0)
        expect(initializeAction).toHaveBeenCalled()
        expect(wrapper).toMatchObject({})
        expect(wrapper).toMatchSnapshot();
    });
});
