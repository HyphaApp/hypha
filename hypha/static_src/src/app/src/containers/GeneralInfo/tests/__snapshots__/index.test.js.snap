// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test General info Container Should render General info without issues 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <GeneralInfoContainer
    initializeAction={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />
</Provider>
`;
